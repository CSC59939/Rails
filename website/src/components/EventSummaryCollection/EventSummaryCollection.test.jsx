import React from 'react';
import renderer from 'react-test-renderer';
import {
  configure,
} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { EventSummaryCollection } from '..';

configure({ adapter: new Adapter() });
const classArray = {
  date: 'Oct. 24',
  events: [
    {
      course: 'CSC 342',
      eventName: 'Review of First Exam',
      dueTime: '3:30pm',
      color: '#d83a42',
      key: 0,
    },
    {
      course: 'CSC 220',
      eventName: 'Homework 1',
      dueTime: '5:00pm',
      color: '#62c4f9',
      key: 1,
    },
  ],
};

// problem need unique key for each iteration
describe('EventSummaryCollection', () => {
  it('Render EventSummaryCollection without any eror', () => {
    expect(renderer.create(<EventSummaryCollection event={classArray} />)
      .toJSON()).toMatchSnapshot();
  });
});
