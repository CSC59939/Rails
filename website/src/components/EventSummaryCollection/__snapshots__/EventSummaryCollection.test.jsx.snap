// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EventSummaryCollection Render EventSummaryCollection without any eror 1`] = `
<div
  className="eventWrapper"
>
  <span
    className="eventDate"
  >
    Oct. 24
  </span>
  <div
    className="events"
  >
    <div
      className="EventSummary"
      style={
        Object {
          "borderColor": "#d83a42",
          "color": "#d83a42",
        }
      }
    >
      <span
        className="course"
      >
        CSC 342
      </span>
      <span
        className="eventName"
      >
        Review of First Exam
      </span>
      <span
        className="dueDate"
      >
        3:30pm
      </span>
    </div>
    <div
      className="EventSummary"
      style={
        Object {
          "borderColor": "#62c4f9",
          "color": "#62c4f9",
        }
      }
    >
      <span
        className="course"
      >
        CSC 220
      </span>
      <span
        className="eventName"
      >
        Homework 1
      </span>
      <span
        className="dueDate"
      >
        5:00pm
      </span>
    </div>
  </div>
</div>
`;
